zjit/src/hir.rs

```diff
@@ -2829,6 +2834,17 @@ impl Function {
                             // entered the compiler.  That means we can just return nil for this
                             // shape + iv name
                             self.push_insn(block, Insn::Const { val: Const::Value(Qnil) })
+                        } else if !recv_type.flags().is_t_object() && !unsafe { rb_jit_multi_ractor_p() } {
```

Comment:
I need to fix this because we shouldn't go to the branches below if `rb_jit_multi_ractor_p()` is true